cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1098,6 +1098,19 @@ def right_scheduler_schema(self, data, **kwargs):
                     f"Scheduling > *Queues section is not appropriate to the Scheduler: {configured_scheduler}."
                 )
 
+    @validates_schema
+    def same_subnet_in_different_queues(self, data, **kwargs):
+        """Validate subnet_ids configured in different queues are the same."""
+        for queues in ["slurm_queues", "aws_batch_queues"]:
+            if data.get(queues, None):
+                queues_list = data.get(queues)
+                subnet_ids_list = []
```

Comment:
if you make `subnet_ids` a `set` you can check `if len(subnet_ids) > 1` in the condition below (`if not all ...`) which would be shorter / read cleaner.

Reply:
Would also be possible to do this all with a set-comprehension...

```.py
    def queue_has_subnet_ids(queue)
        return queue.networking and queue.networking.subnet_ids
    subnet_ids = {q.networking.subnet_ids for q in data.get(queues) if queue_has_subnet_ids(q)}
    if len(subnet_ids) > 1:
        raise ValidationError("SubnetIds configured in different queues should be the same.")
```